src/py_strapi/models/config.py

```
@@ -123,12 +123,38 @@ class StrapiConfig(BaseSettings):
+
+        Raises:
+            ValueError: If URL is not a valid HTTP(S) URL
+        """
+        if not isinstance(v, str):
+            raise ValueError("base_url must be a string")
+
+        url_str = v.strip().rstrip("/")
+
+        # Validate URL format
+        if not url_str:
+            raise ValueError("base_url cannot be empty")
```

Ruff (0.14.14)

[warning] Avoid specifying long messages outside the exception class

(TRY003)